code/game/machinery/doors/firedoor.dm

```diff
@@ -78,17 +82,153 @@
+
+		number_of_people_trying_to_open = max(0, --number_of_people_trying_to_open)
+		user.move_resist = initial(user.move_resist)
+
+/obj/machinery/door/firedoor/attack_paw(mob/living/user, list/modifiers)
+	. = ..()
+	if(!user.combat_mode)
+		attack_hand(user, modifiers)
+
+///deals burn damage to the user depending on whether theyre resistant to heat and how hot the door is, used as a callback in attack_hand's dynamic_do_after
+/obj/machinery/door/firedoor/proc/burn_arms(mob/living/user)
+	. = TRUE //we dont want to interrupt the do_after
+
+	//figure out how "hot" the door is with the temperature of the turf we are on and the user touching us, remember we dont conduct heat to the other side
+	var/datum/gas_mixture/our_air = return_air()
+	var/our_temperature = our_air.temperature
+	var/datum/gas_mixture/users_air = user.return_air()
+	var/users_temperature = users_air.temperature
+
+	var/heat_of_contact_surface = max(our_temperature, users_temperature)
+
+	if(heat_of_contact_surface < MINIMUM_TEMPERATURE_TO_BURN_ARMS)
+		return
+
+	var/heat_protected = FALSE
+	if(HAS_TRAIT(user, TRAIT_RESISTHEAT) || HAS_TRAIT(user, TRAIT_RESISTHEATHANDS))
+		heat_protected = TRUE
+
+	if(!iscarbon(user))
+		if(!heat_protected)
+			user.adjustFireLoss(0.2, forced = TRUE)
+		return
+
+	var/mob/living/carbon/user_as_carbon = user
+
+	if(user_as_carbon.gloves)
+		var/obj/item/clothing/gloves/gloves_of_user = user_as_carbon.gloves
+
+		if(gloves_of_user.max_heat_protection_temperature)
+			heat_protected = heat_protected ? TRUE : (gloves_of_user.max_heat_protection_temperature >= heat_of_contact_surface)
+
+	if(!heat_protected)
+		var/obj/item/bodypart/right_arm = user_as_carbon.get_bodypart(BODY_ZONE_R_ARM)
+		var/obj/item/bodypart/left_arm = user_as_carbon.get_bodypart(BODY_ZONE_L_ARM)
+
+		if(prob(10))
+			playsound(user_as_carbon, 'sound/effects/wounds/sizzle1.ogg', 70)
+
+		if(right_arm && right_arm.status == BODYPART_ORGANIC && !right_arm.is_pseudopart)
+			right_arm.receive_damage(0, pick(0.1, 0.2))
+		if(left_arm && left_arm.status == BODYPART_ORGANIC && !left_arm.is_pseudopart)
+			left_arm.receive_damage(0, pick(0.1, 0.2))
+
+///used in a callback given to do_after_dynamic to adjust the timer based on how many people are trying to open it barehanded
+/obj/machinery/door/firedoor/proc/adjust_do_after_timer(old_delay, multiplicative_action_slowdown)
+	if(old_delay == (TOOLLESS_OPEN_DURATION_SOLO / max((min(3, number_of_people_trying_to_open) * 0.5), 1)) * multiplicative_action_slowdown)
+		return null
+	else
+		return (TOOLLESS_OPEN_DURATION_SOLO / max((min(3, number_of_people_trying_to_open) * 0.5), 1)) * multiplicative_action_slowdown
+
+/obj/machinery/door/firedoor/proc/decrement_bare_hand_openers(datum/source)
+	SIGNAL_HANDLER
+	number_of_people_trying_to_open = max(0, --number_of_people_trying_to_open)
```

Comment:
```suggestion
	number_of_people_trying_to_open = max(0, number_of_people_trying_to_open - 1)
```

1. Don't use the return value of pre/post increment/decrement. It makes for confusing code.
2. Especially don't if you're immediately setting the variable again later.